ledger/db/src/lib.rs

```diff
@@ -235,7 +264,7 @@ impl Ledger for LedgerDB {
             Err(err) => Err(err),
         }?;
 
-        Ok(BlockData::new(block, contents, signature))
+        Ok(ArchiveBlock::new(block, contents, signature, None))
```

Comment:
This feels hacky to me. If there's a signature and we have an AVR for the signer shouldn't we return it as well?